modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js

```diff
@@ -149,12 +149,29 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
 	const publishDefinition = () => {
 		let alertMessage;
 
-		if (!definitionTitle) {
-			alertMessage = Liferay.Language.get('name-workflow-before-publish');
+		if (!definitionTitle || blockingErrors.errorType !== '') {
+			if (!definitionTitle) {
+				alertMessage = Liferay.Language.get(
+					'name-workflow-before-publish'
+				);
+			}
+			else {
+				switch (blockingErrors.errorType) {
+					case 'emptyField':
+						alertMessage = Liferay.Language.get('enter-a-label');
+						break;
+					case 'duplicated':
+						alertMessage = Liferay.Language.get(
+							'a-node-with-that-id-already-exists'
+						);
+						break;
+					default:
+						alertMessage = Liferay.Language.get('error');
+				}
+			}
 
```

Comment:
Hi @M-IgorFranca could you please check if we can simplify the code above with code below?
if (!definitionTitle) {
	alertMessage = Liferay.Language.get('name-workflow-before-publish');

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
} else if (blockingErrors.errorType !== '') {
	switch (blockingErrors.errorType) {
		case 'emptyField':
			alertMessage = Liferay.Language.get('enter-a-label');
			break;
		case 'duplicated':
			alertMessage = Liferay.Language.get(
				'a-node-with-that-id-already-exists'
			);
			break;
		default:
			alertMessage = Liferay.Language.get('error');
	}

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
}

Reply:
Hello @seltong , that would work too, I tried implementing it this way so I don't repeat this code snippet.

`setAlertMessage(alertMessage);
setAlertType('danger');

setShowAlert(true);`